pkg/cni/cilium/cilium-mirror-chart.sh

```diff
@@ -21,7 +21,7 @@ set -euo pipefail
 
 CHART_SOURCE="${CHART_SOURCE:-https://helm.cilium.io}"
 CHART_NAME=${CHART_NAME:-cilium}
-CHART_VERSION="${CHART_VERSION:-1.14.1}"
+CHART_VERSION="${CHART_VERSION:-1.14.2}"
```

Comment:
Adding a "dummy" comment to the review here - Can you please also make sure that:

1. The changelog entry mentions the CVEs / GitHub advisories fixed by these versions? Cilium is a bit weird about this because they don't mention them in their release notes, but the three most recent entries to https://github.com/cilium/cilium/security should be the ones fixed by 1.13.7 / 1.14.2.
2. You update the `go.mod` dependency on `github.com/cilium/cilium` to match `1.14.2`?

Reply:
Note: https://github.com/kubermatic/kubermatic/pull/12666 already bumped the dependency, forgot about that.